src/shared/hooks/useAccessDataForTable.tsx

```diff
@@ -422,43 +432,94 @@ const fileNameForDistributionItem = (distItem: any, defaultName: string) => {
+    createdAt: resource._createdAt,
+    updatedAt: resource._updatedAt,
+    source: 'studios',
+    type: isArray(resource['@type'])
+      ? resource['@type']
+      : [resource['@type'] ?? ''],
+  };
+};
+
+export const toLocalStorageResources = (resource: Resource): TDataSource[] => {
+  const resourceName = getResourceName(resource);
   try {
-    const resourceName = resource.name ?? resource['@id'] ?? resource._self;
-    localStorageRow = {
-      key: resource._self,
-      _self: resource._self,
-      id: resource['@id'],
-      name: resourceName,
-      project: resource._project,
-      description: resource.description ?? '',
-      createdAt: resource._createdAt,
-      updatedAt: resource._updatedAt,
-      source: 'studios',
-      type: isArray(resource['@type'])
-        ? resource['@type']
-        : [resource['@type'] ?? ''],
-
-      distribution: {
-        contentSize: isArray(resource.distribution)
-          ? sumBy(
-              resource.distribution,
-              distItem => distItem.contentSize?.value ?? 0
-            )
-          : resource.distribution?.contentSize ?? 0,
-        encodingFormat: isArray(resource.distribution)
-          ? resource.distribution.find(distItem => distItem.encodingFormat)
-              ?.encodingFormat ?? ''
-          : resource.distribution?.encodingFormat ?? '',
-        label: isArray(resource.distribution)
-          ? resource.distribution.map(distItem => {
-              return fileNameForDistributionItem(distItem, resourceName);
-            })
-          : fileNameForDistributionItem(resource.distribution, resourceName),
-        hasDistribution: has(resource, 'distribution'),
+    // Case 1 - Resource has no distribution
+    if (isNil(resource.distribution)) {
+      return [{ ...baseLocalStorageObject(resource) }];
+    }
+
+    // Case 2 - Resource has (multiple) distributions in an array
+    if (isArray(resource.distribution)) {
+      // In case distribution is an array we have to store the main resource,
+      // but also an object for every item in the distribution array.
+      const localStorageObjs: TDataSource[] = [];
+
+      // First store an object for the parent resource.
+      localStorageObjs.push({
+        ...baseLocalStorageObject(resource),
+        distribution: {
+          hasDistribution: true,
+          contentSize: 0, // So the size in not doubled
+          encodingFormat: '',
+          label: '',
+        },
+      });
+
+      // Now store an object for each distribution item
+      resource.distribution.forEach((distItem, index) => {
+        localStorageObjs.push({
+          ...baseLocalStorageObject(resource, `${index}`),
+          distribution: {
+            hasDistribution: true, // So, we don't download the distribution twice
+            contentSize:
+              (distItem.contentSize as { value: number })?.value ?? // TODO: Verify
+              distItem.contentSize ??
+              0,
```

Comment:
suggestion: we can use the get fn from lodash, it will give nice and typeless expression `_.get(distItem.contentSize, 'value', null) ?? distItem.contentSize`

Reply:
bon idee! :bow: